compat/mingw.h

```diff
@@ -6,11 +6,6 @@ typedef _sigset_t sigset_t;
 #include <winsock2.h>
 #include <ws2tcpip.h>
 
-/* MinGW-w64 reports to have flockfile, but it does not actually have it. */
```

Comment:
Wait, does `mingw-w64` have `flockfile()` now? Back in the days, I got a link error...

If that is fixed now, it would be good to find out exactly when it stopped being an issue, i.e. what value(s) to test `__MINGW64_VERSION_*` for.

Reply:
It doesn't, but it doesn't declare it either. I searched through mingw-w64 history, and it never had flockfile.

What I assume happened is that the compat header did not implement it, and C files only warn about missing declarations, but don't fail the compilation (like the warnings this commit fixes). Then it failed on link time, because the function doesn't really exist.